test/packet/scripts/add_vagrant_box.sh

```diff
@@ -166,13 +169,13 @@ for box in $boxes; do
             url="https://vagrantcloud.com/cilium/boxes/$box/versions/$version/providers/virtualbox.box"
             $aria2c -d "$outdir" -o package.box "$url"
             vagrant box add "cilium/$box" "$outdir/package.box"
-            mkdir -p $box_dir$box
+            mkdir -p "$box_dir$box"
             if [[ ! -f $box_dir$box/metadata_url ]] ; then
-                printf "https://vagrantcloud.com/cilium/$box" > $box_dir$box/metadata_url
+                echo "https://vagrantcloud.com/cilium/$box" > "$box_dir$box/metadata_url"
```

Comment:
Could you please elaborate on this change? Asking because I was under the impression that `printf` was more portable than `echo`. Or is it because you just want a newline character at the end of the file?

Reply:
Ah, good catch with the newline. I initially switched away from `printf` as there were no arguments being passed, and correction functionality required `$box` to not contain any printf verbs (e.g. if `$box` was `foo%s` then `printf` would fail, of course this would not likely happen in practice).

The relevant ShellCheck is [SC3037](https://github.com/koalaman/shellcheck/wiki/SC3037), which indeed warns that flags to `echo` are undefined in POSIX `sh`. However, this script explicitly uses `/bin/bash`, and `-n` is defined there.

Result: I've added a `-n` to `echo`.

Reply:
Ok, I was just wondering why the switch. I understand the `%s` reason. Thanks!

I don't think the newline matters at all, and I think we'd be as well without this `-n` option. It's true we enforce bash so we should have it, but it's just not clear when reading the script what we need to omit the new line (I suspect `printf` didn't add it just because nobody cared to add it). But that's a minor details anyway, so feel free to revert the `-n` or to keep it :).

Reply:
Vagrant itself seems to omit the trailing newline, so let's keep the `-n`.